deploy/docker/Dockerfile

```diff
@@ -19,6 +18,9 @@ RUN apk add git \
     libc-dev \
     libstdc++
 
+COPY tmp/qemu-$QEMU_ARCH-stati* /usr/bin/
+COPY . /emqx-rel
+
```

Comment:
Sort the build stages: run `apk add` first for more efficient cache.

Reply:
Hi, @PMExtra sorry for the delay
We need `qemu` to help us compile other cpu architecture  images on x86, so the `COPY tmp/qemu-$QEMU_ARCH-stati* /usr/bin/` statement needs to be above, You can check https://github.com/multiarch/qemu-user-static

Reply:
@zhanghongtong Thanks for your guidance. But I have some questions.

1. Why don't we use qemu-user-static like:
```
FROM multiarch/qemu-user-static:${QEMU_ARCH} AS qemu
FROM ${BUILD_FROM} AS builder
COPY --from=qemu /usr/bin/qemu-$QEMU_ARCH-static /usr/bin
```

2. Why should we copy `qemu-$QEMU_ARCH-stati*` rather than `qemu-$QEMU_ARCH-static`

Are these all designed to adapt x86_64? (copy nothing if QEMU_ARCH=x86_64)

Reply:
Hi, @PMExtra 

> Why don't we use qemu-user-static like:
> ```
> FROM multiarch/qemu-user-static:${QEMU_ARCH} AS qemu
> FROM ${BUILD_FROM} AS builder
> COPY --from=qemu /usr/bin/qemu-$QEMU_ARCH-static /usr/bin
> ```

The `multiarch/qemu-user-static` image requires the `—privileged` option, so we did not choose it

> Why should we copy qemu-$QEMU_ARCH-stati* rather than qemu-$QEMU_ARCH-static

This is a historical code, it seems there is no difference between the two

> Are these all designed to adapt x86_64? 

Yes, we will build arm, arm64, 386 and s390 docker images on x86_64 machines, you can see them in [GitHub actions](https://github.com/emqx/emqx-rel/pull/616/checks)

Reply:
@zhanghongtong 
抱歉我的英语可能没法准确表达我的意思，暂时中文沟通。
第一个问题我想问的是：目前我们应该是在make脚本中下载qemu的依赖到tmp目录，然后build时从tmp目录复制进去。但我们为什么不直接从`multiarch/qemu-user-static`复制依赖呢？README中也介绍过这种使用方法，这个应该和`--privileged`没有关系的，只是从其中复制文件，并不需要运行容器。
第二个问题我觉得差异在于，使用globs通配符，当文件不存在时会被忽略，不会报错。而直接写完整路径，如果文件不存在，会build失败。
想到这里我怀疑之所以有这两个设计，是不是为了把复制qemu依赖变成一个可选步骤。因为在x86_64上构建x86_64的镜像，其实是不需要qemu的，事实上，也没有`multiarch/qemu-user-static:x86_64`这样的镜像。
所以如果按照我的说法来做，就需要对 x86_64 和其它需要qemu的架构，分开成两个 Dockerfile 文件。而这两个看似有缺点的设计，却能够让所有架构包括x86_64自身都使用同样的Dockerfile来构建。
以上是我的想法，只是想核实一下我的理解是否正确。请不吝赐教。
另外，我已经提交commit把qemu的顺序提到 apk add 前面了，如果没有别的问题，这个PR应该可以合并了。

Reply:
@zhanghongtong Docker CI checks are all success. Please let me know if anything I can do.

Reply:
Hi, @PMExtra 

> 第一个问题我想问的是：目前我们应该是在make脚本中下载qemu的依赖到tmp目录，然后build时从tmp目录复制进去。但我们为什么不直接从multiarch/qemu-user-static复制依赖呢？README中也介绍过这种使用方法，这个应该和--privileged没有关系的，只是从其中复制文件，并不需要运行容器。

I think this is a good idea, can you help us verify that this is work? If successful, we don’t need to download qemu anymore

> 第二个问题我觉得差异在于，使用globs通配符，当文件不存在时会被忽略，不会报错。而直接写完整路径，如果文件不存在，会build失败。

I have tried this point, even if wildcards are used, an error will still be reported when the file does not exist